### PR TITLE
[jit] dropout symbolic_script should respect the training flag

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#20760 [jit] dropout symbolic_script should respect the training flag**

as title

Differential Revision: [D15486511](https://our.internmc.facebook.com/intern/diff/D15486511)

### CODE REVIEW

**Comment by ailzhang** [comment]
is it possible to add a test to prevent regression in the future?

File: torch/csrc/jit/symbolic_script.cpp [hunk 1]
```diff
@@ -1116,18 +1116,29 @@ const std::vector<std::string> functions = {
...
+                    mask = torch.empty_like(input)
+                    mask.bernoulli_(p1m)
+                    res = mask * input / p1m
             else:
-                mask = torch.empty_like(input)
-                mask.bernoulli_(p1m)
-                res = mask * input / p1m
+                p1m = 1.
+                res = input
+                mask = torch.zeros(0)
```
Comment (zdevito) [inline torch/csrc/jit/symbolic_script.cpp]: The definition of backwards when is_training = False is not correct.

Comment (suo) [reply torch/csrc/jit/symbolic_script.cpp]: oops this got merged out somehow

**Review by zdevito — APPROVED** [review]
